functions/src/api/lib/restrictAccessToRole.ts

```diff
@@ -0,0 +1,18 @@
+import {Next} from 'koa';
+import {getAuth} from 'firebase-admin/auth';
+import {ROLES} from '../../../../shared/src/types/User';
+import {FirebaseAuthContext} from './firebaseAuth';
+
+const restrictAccessToRole =
+  (role: keyof typeof ROLES) =>
+  async (ctx: FirebaseAuthContext, next: Next) => {
+    const customClaims = (await getAuth().getUser(ctx.user.id)).customClaims;
```

Comment:
Can't we add the claims on the user context with the existing middleware instead? And keep this one to just check that object?

Reply:
Super improvement, thanks!